lib/sources.nix

```
@@ -229,6 +229,42 @@ let
 
   canCleanSource = src: src ? _isLibCleanSourceWith || !(pathHasContext (toString src));
 
+  /*
+    Get all files (recursively) that pass a filter function. Unlike other source
+    filtering functions, this will not include empty directories in the final
+    derivation.
+
+    Type: sourceLike -> (String -> Bool) -> Source
+
+    Example:
+      filterFiles ./. (lib.hasSuffix ".html")
+  */
+  filterFiles = src: filter:
```

```suggestion
  filterFiles = filter: src:
```

This makes partial application more useful.

---

Oops, I meant to copy the order of `filterSource`, but it seems I swapped them.